### ISSUE TITLE
DatePicker modifiers "disabled"

### ISSUE THREAD

**Issue by fabiocba** [palantir/blueprint#1184]
### Bug report

I need to disable a list of dates according to a function (which finds whether the day is on a list). However, 
Datepicker "disabled" modifiers doesn't disable a Date. 
Any other values for CSS sufix works (if styling it manually) but a check function is necessary to properly disallow clicks. It seems the disabled word is reserved.

- __Package version(s)__: 1.15.0
- __Browser and OS versions__: Firefox 45.9.0 / Chrome 58.0.3029.81 (64-bit) on Debian 3.16.39

#### Steps to reproduce

```
const { DatePicker } = require("@blueprintjs/datetime");

const hasCustomers = (dates) => (date) => {
    return dates.findIndex(d => moment(date).isSame(d, "day")) < 0;  // -1 when not in dates array
}

class MyCalendar extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        // selected and dates are moment objects
        const { dates, selected } = this.props;
        const disabled = hasCustomers(dates); 
        const modifiers = { disabled }; // ANY OTHER NAME HERE WILL DO, BUT EXTRA CSS AND CLICK DISALLOWER NEEDED

        return <DatePicker
                  onChange={(newDate) => this.props.onChange(newDate)}
                  value={selected.toDate()}
                  modifiers={modifiers}
                  localeUtils={LocaleUtils}
                  locale="pt-br" />;
}
module.exports = MyCalendar;
```

#### Actual behavior

```
<div role="row" class="DayPicker-Week">
<div aria-selected="false" aria-disabled="false" aria-label="Dom 18 de Jun de 2017" role="gridcell" tabindex="-1" class="DayPicker-Day">18</div>
<div aria-selected="false" aria-disabled="false" aria-label="Seg 19 de Jun de 2017" role="gridcell" tabindex="-1" class="DayPicker-Day">19</div>
<div aria-selected="false" aria-disabled="false" aria-label="Ter 20 de Jun de 2017" role="gridcell" tabindex="-1" class="DayPicker-Day">20</div>
...

```

#### Expected behavior
```
<div role="row" class="DayPicker-Week">
<div aria-selected="false" aria-disabled="true" aria-label="Dom 18 de Jun de 2017" role="gridcell" tabindex="-1" class="DayPicker-Day DayPicker-Day--disabled">18</div>
<div aria-selected="false" aria-disabled="true" aria-label="Seg 19 de Jun de 2017" role="gridcell" tabindex="-1" class="DayPicker-Day DayPicker-Day--disabled">19</div>
<div aria-selected="false" aria-disabled="true" aria-label="Ter 20 de Jun de 2017" role="gridcell" tabindex="-1" class="DayPicker-Day DayPicker-Day--disabled">20</div>
<div aria-selected="false" aria-disabled="false" aria-label="Qua 21 de Jun de 2017" role="gridcell" tabindex="-1" class="DayPicker-Day DayPicker-Day">21</div>
...
```

**Comment by giladgray** [comment]
@fabiocba `DatePicker` does not support custom `disabled` modifier because it implements its own. [See here](https://github.com/palantir/blueprint/blob/master/packages/datetime/src/datePickerCore.tsx#L59-L65) for the full list of disallowed modifiers. Sorry this isn't spelled out in the docs.

**Comment by fabiocba** [comment]
Hey @giladgray, thank you. That's OK. Just in case someone else has a similar question, I've managed to disable the dates I wanted by using a different name ("disable") for the modifiers property and defining an intermediate function to prevent clicks on undesirable days:

```
...
const disable = hasCustomers(dates); 
const onChange = (newDate) => {
   if (!disable(newDate))
       this.props.onChange(newDate);
};
const modifiers = { disable };

return <DatePicker
         onChange={onChange}
         modifiers={modifiers}
...
```
Also, in CSS I created the disable class using the same properties as in the blueprintjs css:

```
.pt-datepicker .DayPicker-Day--disable
  background: none;
  color: rgba(191, 204, 214, 0.5);
  cursor: not-allowed;
```

Best

**Comment by cmslewis** [comment]
(Circling back to this with a quick thought: could maybe even print a console warning or throw an error if consumers try to pass their own disabled dates.)

**Comment by llorca** [comment]
Resolved with #1647
